src/coord/src/coord.rs

```diff
@@ -1205,8 +1213,14 @@ where
         when: PeekWhen,
         finishing: RowSetFinishing,
         materialize: bool,
+        as_of: Option<u64>,
     ) -> Result<ExecuteResponse, failure::Error> {
-        let timestamp = self.determine_timestamp(&source, when)?;
+        let peek_when = if let Some(as_of) = as_of {
```

Comment:
Reading through, I think this should be equivalent to [the first `if` clause in `determine_frontier`](https://github.com/MaterializeInc/materialize/blob/master/src/coord/src/coord.rs#L2179), which we for sinks and tails.